benchmarks/tests/test_dataset_select_benchmark.py

```diff
@@ -0,0 +1,54 @@
+import copy
+import json
+
+import pytest
+
+from .. import _sources
+from .. import dataset_select_benchmark
+from ..tests._asserts import assert_cli, assert_benchmark
+
+
+HELP = """
+Usage: conbench dataset-select [OPTIONS] SOURCE
+
+  Run dataset-select benchmark.
+
+Options:
+  --cpu-count INTEGER
+  --iterations INTEGER   [default: 1]
+  --gc-collect BOOLEAN   [default: true]
+  --gc-disable BOOLEAN   [default: true]
+  --show-result BOOLEAN  [default: true]
+  --show-output BOOLEAN  [default: false]
+  --run-id TEXT          Group executions together with a run id.
+  --help                 Show this message and exit.
+"""
+
+
+nyctaxi = _sources.Source("nyctaxi_multi_parquet_s3_repartitioned")
+benchmark = dataset_select_benchmark.DatasetSelectBenchmark()
+
+
+def assert_run(run, index, benchmark, source):
+    result, output = run[index]
+    assert_benchmark(result, source.name, benchmark.name)
+    print(json.dumps(result, indent=4, sort_keys=True))
+    assert "pyarrow.Table" in str(output)
+
+
+def test_dataset_read_one():
```

Comment:
These 2 tests fail for me.

```
    def test_dataset_read_one():
>       [(result, output)] = benchmark.run(nyctaxi, iterations=1)

/Users/diana/workspace/benchmarks/benchmarks/tests/test_dataset_select_benchmark.py:40: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/Users/diana/workspace/benchmarks/benchmarks/dataset_select_benchmark.py:52: in run
    dataset = pyarrow.dataset.dataset(
/Users/diana/envs/qa/lib/python3.9/site-packages/pyarrow/dataset.py:616: in dataset
    return _filesystem_dataset(source, **kwargs)
/Users/diana/envs/qa/lib/python3.9/site-packages/pyarrow/dataset.py:380: in _filesystem_dataset
    return factory.finish(schema)
pyarrow/_dataset.pyx:1700: in pyarrow._dataset.DatasetFactory.finish
    ???
pyarrow/error.pxi:122: in pyarrow.lib.pyarrow_internal_check_status
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   OSError: When reading information for key '2009/01/0000/data.parquet' in bucket 'ursa-labs-taxi-data-repartitioned-10k': AWS Error [code 15]: No response body.

pyarrow/error.pxi:99: OSError

```

Reply:
Sorry, I hadn't properly made the bucket public before. It should be OK now (I tested after clearing my AWS credentials).